rocAL/rocAL/source/pipeline/master_graph.cpp

```diff
@@ -460,7 +500,462 @@ MasterGraph::timing()
+                        {
+                            float *output_tensor_32 = static_cast<float *>(out_ptr);
+                            auto channel_size = w * h;
+                            for (unsigned channel_idx = 0; channel_idx < c; channel_idx++) {
+                                for (unsigned i = 0; i < channel_size; i++)
+                                    output_tensor_32[dest_buf_offset + channel_idx + i * c] =
+                                            offset[channel_idx] + multiplier[channel_idx] *
+                                                                    (reverse_channels ? (float) (in_buffer[i * c + c - channel_idx - 1])
+                                                                                    : (float) (in_buffer[i * c + channel_idx]));
+                            }
+                        }
+                        else if(output_data_type == RocalTensorDataType::FP16)
+                        {
+                            half *output_tensor_16 = static_cast<half *>(out_ptr);
+                            auto channel_size = w * h;
+                            for (unsigned channel_idx = 0; channel_idx < c; channel_idx++) {
+                                for (unsigned i = 0; i < channel_size; i++)
+                                    output_tensor_16[dest_buf_offset + channel_idx + i * c] =
+                                            offset[channel_idx] + multiplier[channel_idx] *
+                                                                (reverse_channels ? (half) (in_buffer[i * c + c - channel_idx - 1])
+                                                                                    : (half) (in_buffer[i * c + channel_idx]));
+                            }
+                        }
+                    }
+                    if(format == RocalTensorlayout::NCHW)
+                    {
+                        if(output_data_type == RocalTensorDataType::FP32)
+                        {
+                            float *output_tensor_32 = static_cast<float *>(out_ptr);
+                            auto channel_size  = w * h;
+                            if(c != 3)
+                            {
+                                for(unsigned i = 0; i < channel_size; i++)
+                                    output_tensor_32[dest_buf_offset + i] = offset[0] + multiplier[0]*(float)in_buffer[c*i];
+                            }
+                            else {
+        #if (ENABLE_SIMD && __AVX2__)
+                                float *B_buf = output_tensor_32 + dest_buf_offset;
+                                float *G_buf = B_buf + channel_size;
+                                float *R_buf = G_buf + channel_size;
+
+                                __m256i mask_B, mask_G, mask_R;
+                                if (reverse_channels) {
+                                    mask_B = _mm256_setr_epi32(0x80808000, 0x80808003, 0x80808006, 0x80808009, 0x80808000,
```

Comment:
Please move this masks outside in a common file, instead of setting for each image
This will save some cycles